google/api_core/path_template.py

```diff
@@ -193,3 +193,58 @@ def validate(tmpl, path):
     """
     pattern = _generate_pattern_for_template(tmpl) + "$"
     return True if re.match(pattern, path) is not None else False
+
+def transcode(http_options, **request_kwargs):
+    """Transcodes a grpc request pattern into a proper HTTP request following the rules outlined here,
+       https://github.com/googleapis/googleapis/blob/master/google/api/http.proto#L44-L312
+
+        Args:
+            http_options (list(dict)): A list of dicts which consist of these keys,
+                'method'    (str): The http method
+                'uri'       (str): The path template
+                'body'      (str): The body field name (optional)
+                (This is a simplified representation of the proto option `google.api.http`)
+            request_kwargs (dict) : A dict representing the request object
+
+        Returns:
+            dict: The transcoded request with these keys,
```

Comment:
I think it would be useful to include, in the return, the actual rule that matched, so somewhere below (say before line 248), you would have:
request['matched_rule'] = http_option

This would help the calling template figure out how to represent the returned body.  The body can't be used as-is because the names are snake case, and the actual json body needs to camel case, so the caller needs to manipulate it.

Reply:
As per offline discussion, not including this change in this pull request.